py/picca/test/test_3_cor.py

```diff
@@ -0,0 +1,645 @@
+'''
+Test module
+'''
+
+import unittest
+import numpy as np
+import fitsio
+import healpy
+import os
+import tempfile
+import shutil
+from pkg_resources import resource_filename
+import sys
+
+from picca.utils import userprint
+
+from .test_helpers import update_system_status_values, compare_fits, compare_h5py, send_requirements, load_requirements
+import importlib
+
+
+class TestCor(unittest.TestCase):
+    #TODO: bad style, using it for the moment while transitioning, remove later
+    compare_fits = compare_fits
+    compare_h5py = compare_h5py
+
+    @classmethod
+    def setUpClass(cls):
+        cls._branchFiles = tempfile.mkdtemp() + "/"
+        cls.produce_folder(cls)
+        cls.picca_base = resource_filename('picca',
+                                           './').replace('py/picca/./', '')
+        send_requirements(load_requirements(cls.picca_base))
+        np.random.seed(42)
+        cls._masterFiles = cls.picca_base + '/py/picca/test/data/'
+        cls._test=True
+        userprint("\n")
+
+    @classmethod
+    def tearDownClass(cls):
+        if os.path.isdir(cls._branchFiles):
+            shutil.rmtree(cls._branchFiles, ignore_errors=True)
+
+
+    def produce_folder(self):
+        """
+            Create the necessary folders
+        """
+
+        userprint("\n")
+        lst_fold = [
+            "/Products/",
+            "/Products/Spectra/",
+            "/Products/Correlations/",
+            "/Products/Correlations/Co_Random/",
+            "/Products/Correlations/Fit/",
+        ]
+
+        for fold in lst_fold:
+            if not os.path.isdir(self._branchFiles + fold):
+                os.mkdir(self._branchFiles + fold)
+
+        return
+
+    def test_cf1d(self):
+        import picca.bin.picca_cf1d as picca_test
```

Comment:
imports should be placed at the top of the file. Also, do not name all the imports with the same name. This way we won't need the reload and it will be much clearer which function is being called. The same comment stands for the rest of the functions of this package

Reply:
I did fix the imports. However, we'll still need some of the reloads as global variables in `picca.(x)cf` will be overwritten during the test making subsequent tests fail. Previously this has been circumvented by running all tests through subprocesses, but that gives less insight into where tests actually fail